tools/utils.cpp

```diff
@@ -330,23 +326,15 @@ libff::alt_bn128_G1 ThresholdUtils::HashtoG1(
             result.X = x1;
             libff::alt_bn128_Fq temp_y = y1_sqr.sqrt();
 
-            mpz_t pos_y;
-            mpz_init( pos_y );
-
-            temp_y.as_bigint().to_mpz( pos_y );
-
-            mpz_t neg_y;
-            mpz_init( neg_y );
+            mpz_class y;
+            temp_y.as_bigint().to_mpz( y.get_mpz_t() );
 
-            ( -temp_y ).as_bigint().to_mpz( neg_y );
+            mpz_class neg_y = -y;
```

Comment:
please change to neg_y = -y % q here, otherwise the next line will always return false